app/javascript/packs/receiver.jsx

```diff
@@ -15,44 +16,76 @@ export default class Receiver extends React.PureComponent {
     this.TRIGGER_INTERVAL = 'TRIGGER_INTERVAL';
 
 
-    console.log(props);
-    this.state = {
-      selectedTriggerOption: this.TRIGGER_SCHEDULED
-    };
-
-    this.updateScheduleTime = this.updateScheduleTime.bind(this);
-    this.selectTriggerOption = this.selectTriggerOption.bind(this);
-    this.updateInterval = this.updateInterval.bind(this);
-    this.updateStartAtTime = this.updateStartAtTime.bind(this);
-    this.cancelUpdate = this.cancelUpdate.bind(this);
-    this.save = this.save.bind(this);
-    this.delete = this.delete.bind(this);
+    this.state = this.nullState();
+
+    [
+      'updateType',
+      'updateStartTime',
+      'cancelUpdate',
+      'save',
+      'delete'
+    ].forEach(fn => this[fn] = this[fn].bind(this));
```

Comment:
not `.bind` 😭

Reply:
:')

Back in the early days of React this happened automatically, but when they piggybacked off of ES6 classes they had to keep the default behaviour of unbound methods. So now you have to bind a bunch of stuff yourself, rip

Reply:
I thought the point of the frameworks was to abstract away the bindings 😢 sad reacc only